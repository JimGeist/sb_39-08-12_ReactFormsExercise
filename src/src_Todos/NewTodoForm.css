label {
    margin: 10px 5px 10px 5px;
}

input {
    margin-right: 20px; 
    font-weight: bold;
    width: 20.0rem;
}

.NewTodoForm-BtnAdd {
    display: block;
    padding: 3px 10px 3px 10px;
    margin: 15px 0 15px 5px;
}