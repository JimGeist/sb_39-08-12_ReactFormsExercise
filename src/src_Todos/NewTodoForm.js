import React, { useState } from "react";

import "./NewTodoForm.css";

function NewTodoForm({ addTodoFx }) {

    const INITIAL_DATA = { todo: "" };

    const [formData, setFormData] = useState(INITIAL_DATA);

    const handleChange = (evt) => {

        const { name, value } = evt.target;

        setFormData(formData => (
            { ...formData, [name]: value }
        )
        )
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();

        // Date.now() is the number of milliseconds since Jan. 1, 1970.
        //  Personally, a bit more manageable key than the one uuid creates.
        addTodoFx({ ...formData, id: Date.now() });
        setFormData(INITIAL_DATA);
    }

    return (
        <form className="NewTodoForm" onSubmit={handleSubmit} >
            <label htmlFor="todo">New todo:</label>
            <input
                name="todo"
                id="todo"
                type="text"
                className="NewTodoForm-InpTodo"
                value={formData.todo}
                onChange={handleChange}
            />
            <button className="NewTodoForm-BtnAdd" >Add New Todo</button>
        </form>
    )

}

export default NewTodoForm;