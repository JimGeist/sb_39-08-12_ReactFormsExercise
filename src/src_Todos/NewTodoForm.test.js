import React from "react";
import { render, screen } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// Smoke Test -- does NewTodoForm render?
it("renders the NewTodoForm component without crashing", () => {
    render(<NewTodoForm />);
});

// NewTodoForm snapshot tests
it("NewTodoForm component matches snapshot", () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("has a 'New todo:' label and an 'Add New Todo' button", function () {
    const { getByLabelText, getByText } = render(<NewTodoForm />);

    // expect 'New todo:' label
    expect(getByLabelText("New todo:")).toBeInTheDocument();

    // expect 'Add New Todo' button
    expect(getByText("Add New Todo")).toBeInTheDocument();

});