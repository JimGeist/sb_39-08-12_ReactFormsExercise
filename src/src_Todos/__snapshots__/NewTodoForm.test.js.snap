// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="todo"
    >
      New todo:
    </label>
    <input
      class="NewTodoForm-InpTodo"
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-BtnAdd"
    >
      Add New Todo
    </button>
  </form>
</DocumentFragment>
`;
